atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -187,8 +188,16 @@ public CompletableFuture<InstallResponse> onInstall(final InstallRequest request
                     .build()));
       }
 
-      pendingSnapshot =
-          raft.getPersistedSnapshotStore().newReceivedSnapshot(snapshotChunk.getSnapshotId());
+      try {
+        pendingSnapshot =
+            raft.getPersistedSnapshotStore().newReceivedSnapshot(snapshotChunk.getSnapshotId());
+      } catch (final SnapshotAlreadyExistsException snapshotAlreadyExists) {
+        // This should not happen because we previously check for the latest snapshot. But, if it
+        // happens, instead of crashing raft thread, we respond with success because we already have
+        // the snapshot.
+        return CompletableFuture.completedFuture(
+            logResponse(InstallResponse.builder().withStatus(RaftResponse.Status.OK).build()));
```

Comment:
❓ 🧪 Can we have a test for such? That if we send multiple times the same install request that we get immediately an success for example ? 🤔

Reply:
This case would be covered randomized tests for liveness, right? Eventually, the follower that received the snapshot also receives the following log entries. If it doesn't liveness test fails.
We can add a test in `PassiveRoleTest`, but I'm not keen on that because we have to mock everything to be able to test a small thing.